### PR TITLE
Update AdjustableForKeyboard.swift

### PR DESCRIPTION
**Pull request by NSMyself**
- Updated to Swift 4.2
- Changed the method names in the protocol
- Fix the remove all observers bug

### CODE REVIEW

File: AdjustableForKeyboard.swift [hunk 1]
```diff
@@ -14,32 +14,39 @@ extension AdjustableForKeyboard where Self: UIViewController {
...
+        
+        guard
+            let info = notification.userInfo,
+            let keyboardFrame = info[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect else {
+                return
         }
         
-        self.bottomConstraint.constant = -(keyboardFrame.size.height + 8)
-        self.view.layoutIfNeeded()
+        self.bottomConstraint.constant = -(keyboardFrame.size.height + self.offset)
```
Comment (phelgo) [inline AdjustableForKeyboard.swift:43]: Minor detail: since we are at it, we can remove `self` here too

**Review by phelgo — APPROVED** [review]
👍

File: AdjustableForKeyboard.swift [hunk 2]
```diff
@@ -1,11 +1,11 @@
-import Foundation
+
 import UIKit
 
-protocol AdjustableForKeyboard {
+protocol AdjustableForKeyboard: class {
```
Comment (nunojfg) [inline AdjustableForKeyboard.swift:4]: Why class here?

**Review by nunojfg — APPROVED** [review]
Seems fine to me will accept the MR :)

File: AdjustableForKeyboard.swift (hunk 2 above)
Comment (NSMyself) [reply AdjustableForKeyboard.swift:4]: Since you're using this explicitly on UIViewControllers it makes sense to restrict this protocol to classes.

**Pull request merged**